packages/angular_devkit/build_angular/src/tools/esbuild/stylesheets/sass-language.ts

```diff
@@ -81,6 +81,12 @@ async function compileString(
     sassWorkerPool = new sassService.SassWorkerImplementation(true);
   }
 
+  // Cache is currently local to individual compile requests.
+  // Caching follows Sass behavior where a given url will always resolve to the same value
+  // regardless of its importer's path.
+  const resolutionCache = new Map<string, URL | null>();
```

Comment:
Question: Is there a need for a distinct `null` in this `Map` as distinct from `undefined` (not in the `Map`)?

What does it _mean_ to have a URL resolve to `null` as opposed to `undefined`? Is it for the error case where the URL can't be resolved?

Reply:
Returning `null` is not actually an error at this stage.  It just means this importer plugin could not resolve the given input. A later plugin could then successfully resolve it. Configured Sass load paths, for instance, are always tried last. Within that context, if the same input is provided again within a compilation then there's no need to attempt the full resolution when it can instead skip it and let the later stages handle it right away.

Reply:
Ok, so we're caching the not found case. Can you include a comment on the `resolutionCache` declaration to write this down in the code for future readers?